Update env

mmkekic opened the pull request:
This PR updates the environment packages.

mmkekic committed:
Fix fixture usage in paolina_tests

mmkekic committed:
Change deprecated getfuncargvalue to getfixturevalue

mmkekic committed:
Accomodate _decode_str_columns to new pandas version

mmkekic committed:
Convert np.ceil to int for np.linspace to work

gonzaponte commented on conftest.py:
```diff
@@ -7,8 +7,8 @@
-settings.register_profile("hard_nocov", settings(max_examples = 1000, use_coverage=False))
-settings.register_profile("dev_nocov" , settings(max_examples =   10, use_coverage=False))
+settings.register_profile("hard_nocov", settings(max_examples = 1000))
+settings.register_profile("dev_nocov" , settings(max_examples =   10))
```
How can we deactivate coverage now? Or we don't care anymore? In any case, we have duplicated profiles now.

mmkekic replied:
```
settings._define_setting(
    "use_coverage",
    default=not_set,
    deprecation_message="""
use_coverage no longer does anything and can be removed from your settings.
""",
    deprecated_since="2017-09-14",
    description="""
A flag to enable a feature that no longer exists. This setting is present
only for backwards compatibility purposes.
""",
)
```
Seems that we couldn't deactivate for some time already!

gonzaponte commented on manage.sh:
```diff
@@ -57,22 +57,27 @@ function install_conda {
+            echo Conda already installed.
+	else
+	    conda activate base
+	    conda install conda=4.8.2
```
`conda deactivate` after update?

gonzaponte replied:
Another thing that is bothering me is that, as I understand it, the update affects all environments and therefore it has side effects for other environments we may have (I have a few, at least). Is this the case? If so maybe we shouldn't force the conda version. If not then this should be done just for the IC environment.

mmkekic replied:
I dont think there should be any problems with keeping old environments with updated conda version, but maybe is safer to just print a message saying that the conda is outdated and one can update it with a given command

gonzaponte commented on manage.sh:
```diff
@@ -57,22 +57,27 @@ function install_conda {
-        echo Activated conda by sourcing $CONDA_SH
+	CONDA_SH=$HOME/miniconda/etc/profile.d/conda.sh
+	source $CONDA_SH
+	echo Activated conda by sourcing $CONDA_SH
```
why did you dedent? It shouldn't be dedented as it refers to the miniconda autoinstalled.

gonzaponte reviewed (changes requested):
We should have an automated test for this thing. Since we don't we should have a non-automated one at least. Try to run the script both installing and not installing conda. _neutrinos_ would be a good test bench for this as many people will use it there.

mmkekic committed:
Update conda packages

The packages compatible with python 3.8 as well

mmkekic committed:
Remove nocov profiles since use_coverage is deprecated.

mmkekic committed:
Use keyword elements for hypothesis strategies

In future hypothesis version this will be a keyword-only argument

mmkekic committed:
Fix usage of hypothesis strategies

The examples were drawn using .example that is only for interactive purpuses

mmkekic committed:
Add pytest marks for slow and serial

pytest will always raise a warning for custom non-documented marks

mmkekic committed:
Use raw strings in regex searches

mmkekic committed:
Skip tests relying on scipy interpolation.

The scipy function is giving different result. This issue has to be
addressed in the future and the tests recovered.

mmkekic committed:
Change collection deprecationwarning message

mmkekic committed:
Ignore 'dataframe is empty' warning

mmkekic committed:
Ignore UserWarnings in city on empty files

mmkekic committed:
Use isinstance instead of is in for enum members

Using non-Enums in containment checks will raise TypeError in Python 3.8

gonzaponte reviewed (approved):
This PR updates our packages to a recent version, much needed after freezing them for 1.5 years.
There are some caveats:
- Some tests have been skipped due to a change in behavior of scipy.interpolate that doesn't affect the final result. These are to be fixed soon.
- Some tests have moved from a single-input to a hypothesis-generated input, which may increase test-execution time (we haven't evaluated it).

Otherwise, this is ready to be merged. Good job.
